independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapArtifactVersion.java

```diff
@@ -0,0 +1,359 @@
+package io.quarkus.bootstrap.resolver.maven;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TreeMap;
+import org.eclipse.aether.version.Version;
+
+public class BootstrapArtifactVersion implements Version {
```

Comment:
IMO you should use SmallRye Common's version facility.  For no particular reason :grin: